docs/content/contributing/building.md

```diff
@@ -7,34 +7,106 @@ weight: 250
+## Requirements
 
-You should do the following in order to build the containers locally and be able to submit patches:
+- CentOS 7 or Red Hat 7 environment. The instructions below are set up for
+CentOS 7, but you can read the [installation guide](/installation-guide/installation-guide)
+for additional instructions
+- [`go`](https://golang.org/) version 1.13+
+- [`buildah`](https://buildah.io/) 1.9+
+- [`git`](http://git-scm.org/)
 
-1. Fork the [Crunchy-Containers](https://github.com/CrunchyData/crunchy-containers) GitHub repository.
-2. Containers builds are installed via a Makefile. You will need to run the following commands:
+## Setup
 
+1. On your system, be sure you have installed the requirements. You can do so
+with the following commands:
+
+```shell
+sudo yum -y install epel-release
+sudo yum -y install golang git buildah
+```
+
+2. Create a fork of the [Crunchy Container Suite](https://github.com/CrunchyData/crunchy-containers)
+repository and clone it to the environment using the following commands below:
+
+```shell
+mkdir -p $HOME/cdev/src/github.com/crunchydata $HOME/cdev/pkg $HOME/cdev/bin
+cd $HOME/cdev/src/github.com/crunchydata
+git clone https://github.com/crunchydata/crunchy-containers
+cd crunchy-containers
+```
+
+3. Set yup your local environmental variables to reference the containers you
+want to build, For instance, to build the latest version, you would use the
+following variables:
+
+```shell
+export GOPATH=$HOME/cdev        # set path to your new Go workspace
+export GOBIN=$GOPATH/bin        # set bin path
+export PATH=$PATH:$GOBIN        # add Go bin path to your overall path
+export CCP_BASEOS=centos7       # centos7 for Centos, ubi7 for Red Hat Universal Base Image
+export CCP_PGVERSION=12        # The PostgreSQL major version
+export CCP_PG_FULLVERSION=12.2
+export CCP_VERSION=4.3.0
+export CCP_IMAGE_PREFIX=crunchydata # Prefix to put before all the container image names
+export CCP_IMAGE_TAG=$CCP_BASEOS-$CCP_PG_FULLVERSION-$CCP_VERSION   # Used to tag the images
+export CCPROOT=$GOPATH/src/github.com/crunchydata/crunchy-containers    # The base of the clone github repo
+```
+
+You can save these variables to be set each time you open up your shell by
+adding them to your `.bashrc` file:
+
+```shell
+cat >> ~/.bashrc <<-EOF
+export GOPATH=$HOME/cdev
+export GOBIN=$GOPATH/bin
+export PATH=$PATH:$GOBIN
+export CCP_BASEOS=centos7
+export CCP_PGVERSION=12
+export CCP_PG_FULLVERSION=12.2
+export CCP_VERSION=4.3.0
+export CCP_IMAGE_PREFIX=crunchydata
+export CCP_IMAGE_TAG=$CCP_BASEOS-$CCP_PG_FULLVERSION-$CCP_VERSION
+export CCPROOT=$GOPATH/src/github.com/crunchydata/crunchy-containers
+EOF
+```
+
+4. Download the GPG Keys and repository files that are required to pull in the
+packages used to build the containers:
+
+```shell
+cd $CCPROOT
+curl https://api.developers.crunchydata.com/downloads/repo/rpm-centos/postgresql12/crunchypg12.repo > conf/crunchypg12.repo
```

Comment:
Investigate question - is libcurl on by default on centos7?  I know that wget is not, but for curl it may be worth a double-check.

Reply:
It appears to be. It's also on by default in MacOS